Add .asin() method to Lua documentation (#3766)
Codecademy/docs: content/lua/concepts/mathematical-library/terms/asin/asin.md

---
Title: '.asin()'
Description: 'Returns the inverse sine in radians of a given value.'
Subjects:
  - 'Computer Science'
  - 'Game Development'
Tags:
  - 'Trigonometry'
  - 'Functions'
  - 'Math'
  - 'Numbers'
CatalogContent:
  - 'learn-lua'
  - 'paths/computer-science'
---

Lua's **`math.asin()`** function calculates the inverse sine, also known as arcsine, of a given value. In other words, it finds the angle in radians whose sine equals the given value. The function takes a number within the range of `-1` to `1` as input, representing a sine value, and returns the corresponding angle in radians. This is particularly useful in various applications, such as physics simulations and game development, where understanding angles and trajectories is essential for tasks like character movement and object positioning.

> > **Note**: Understanding the inner workings of `math.asin()` is essential for anyone in the programming field. While specific tasks can often be achieved using libraries and frameworks, a solid grasp of this function's functionality is fundamental. This knowledge not only helps in solving immediate problems but also empowers programmers to create new applications and innovate within their respective domains.

Reviewer: ```suggestion
```

Reviewer: This note is not really bringing attention to a specific aspect of the functionality, or a caveat to be aware of. It's more of a general recommendation how understanding the implementation can have broader benefits in understanding other programming problems.

## Syntax

The `math.asin()` function must be called with an argument using the following syntax:

```pseudo
math.asin(x)
```

> - Where `x` is a number between - 1 and 1, representing a sine value for finding the corresponding angle in radians. If no argument is passed, the program will raise an error.

Reviewer: ```suggestion
- Where `x` is a number between `-1` and `1`, representing a sine value for finding the corresponding angle in radians. If no argument is passed, the program will raise an error.
```

## Example 1

The following example shows the usage of `math.asin()`:

```lua
print(math.asin(1))
```

Output:

```shell
1.5707963267949
```

> ## Example 2

Reviewer: This example doesn't seem quite right. Generally, projectile problems (whether time of flight, max height, etc) all include a gravitational component (g), which is not present here. I would suggest deleting the example for the sake of simplification

---
Title: 'asin()'
Description: 'Returns the inverse sine in radians of a given value.'
Subjects:
  - 'Computer Science'
  - 'Game Development'
Tags:
  - 'Trigonometry'
  - 'Functions'
  - 'Math'
  - 'Numbers'
CatalogContent:
  - 'learn-lua'
  - 'paths/computer-science'
---

Lua's `math.asin()` function calculates the inverse sine, also known as arcsine, of a given value. In other words, it finds the angle in radians whose sine equals the given value. The function takes a number between -1 and 1 as input and returns the corresponding angle in radians. This is particularly useful in various applications, such as physics simulations and game development, where understanding angles and trajectories is essential.

> > **Note**: Mastering the intricacies of `math.asin()` is crucial for anyone in the programming field. Although specific tasks might be accomplished through libraries and frameworks, a solid understanding of how the function works is fundamental. It not only aids in solving immediate problems but also empowers programmers to develop new applications and innovate in their domains.

Reviewer 2: ```suggestion
> **Note**: Understanding the inner workings of `math.asin()` is essential for anyone in the programming field. While specific tasks can often be achieved using libraries and frameworks, a solid grasp of this function's functionality is fundamental. This knowledge not only helps in solving immediate problems but also empowers programmers to create new applications and innovate within their respective domains.
```

## Syntax

The `math.asin()` function must be called with an argument using the following syntax:

```pseudo
math.asin(x)
```

- Where `x` is a number between - 1 and 1. If no argument is passed, the program will raise an error.

## Example 1

The following example shows the usage of `math.asin()`:

```lua
print(math.asin(1))
```

Output:

```shell
1.5707963267949
```

## Example 2

The Lua script below provides a deeper example of `math.asin()`. It calculates and prints the launch angle in degrees required to hit a target 15 meters away using an arrow traveling at a speed of 10 meters per second, demonstrating the use of the `math.asin()` function in trigonometric calculations for projectile motion.

```lua

> -- Fixed values

Reviewer 2: ```suggestion
```

local arrow_speed = 10 -- speed of the arrow in meters per second
local target_distance = 15 -- distance to the target in meters

> -- Calculating launch angle using simplified formula

Reviewer 2: ```suggestion
```
